spring-core/src/main/java/org/springframework/util/LinkedCaseInsensitiveMap.java

```diff
@@ -119,11 +119,10 @@ public boolean containsKey(Object key) {
 			}
 			@Override
 			protected boolean removeEldestEntry(Map.Entry<String, V> eldest) {
-				boolean doRemove = LinkedCaseInsensitiveMap.this.removeEldestEntry(eldest);
-				if (doRemove) {
+				if (LinkedCaseInsensitiveMap.this.removeEldestEntry(eldest)) {
```

Comment:
Please revert the changes to the `removeEldestEntry()` method, since the proposal changes the logic.